Python/Product/TestAdapter/TestContainerDiscoverer.cs

```diff
@@ -167,330 +135,121 @@ private TestContainerDiscoverer([Import(typeof(SVsServiceProvider))]IServiceProv
+                    // Raise containers changed event...
+                    ContainersChanged();
                 }
             }
 
-            OnTestContainersChanged(e.Project);
-        }
-
-        private void OnProjectRenamed(object sender, ProjectEventArgs e) {
-            OnProjectUnloaded(this, e);
-            OnProjectLoaded(this, e);
-        }
-
-        /// <summary>
-        /// Handler to react to test file Add/remove/rename events
-        /// </summary>
-        private void OnProjectItemChanged(object sender, TestFileChangedEventArgs e) {
-            if (e != null && ShouldDiscover(e.File)) {
-                string root = null;
-                switch (e.ChangedReason) {
-                    case TestFileChangedReason.Added:
-                        Debug.Assert(e.Project != null);
-                        if (e.Project.IsTestProject(GuidList.guidPythonProjectGuid)) {
-                            root = e.Project.GetProjectHome();
-
-                            if (!string.IsNullOrEmpty(root) && CommonUtils.IsSubpathOf(root, e.File)) {
-                                _testFilesUpdateWatcher.AddDirectoryWatch(root);
-                                _fileRootMap[e.File] = root;
-                            } else {
-                                _testFilesUpdateWatcher.AddWatch(e.File);
-                            }
-
-                            OnTestContainersChanged(e.Project);
-                        }
-                        break;
-                    case TestFileChangedReason.Removed:
-                        Debug.Assert(e.Project != null);
-
-                        if (_fileRootMap.TryGetValue(e.File, out root)) {
-                            _fileRootMap.Remove(e.File);
-                            if (!_fileRootMap.Values.Contains(root)) {
-                                _testFilesUpdateWatcher.RemoveWatch(root);
-                            }
-                        } else {
-                            _testFilesUpdateWatcher.RemoveWatch(e.File);
-                        }
-
-                        // https://pytools.codeplex.com/workitem/1546
-                        // track the last delete as an update as our file system scan won't see it
-                        _lastWrite = DateTime.Now.ToUniversalTime();
-
-                        OnTestContainersChanged(e.Project);
-                        break;
-                    // VS renames files instead of overwriting them when
-                    // saving, so we need to listen for renames where the new
-                    // path is part of the project.
-                    case TestFileChangedReason.Renamed:
-                    case TestFileChangedReason.Changed:
-                        OnTestContainersChanged(GetTestProjectFromFile(e.File));
-                        break;
+            private Architecture Architecture {
+                get {
+                    return Architecture.X86;
```

Comment:
I suspect we can switch to Any here and then we'll work with 64-bit tests, but we can try that later.